custom_components/pyscript/__init__.py

```diff
@@ -250,6 +266,55 @@ async def unload_scripts(global_ctx_only=None, unload_all=False):
         await GlobalContextMgr.delete(global_ctx_name)
 
 
+@bind_hass
+async def install_requirements(hass):
```

Comment:
This function does I/O and could be quite slow installing requirements.  So it has to run outside the main event loop.  It should be run in another thread via `hass.async_add_executor_job`.

Reply:
I can make that change, but are there any concerns that we will load files and attempt to run them/import packages before requirements are installed?

Reply:
nevermind, we would still await it. I will submit a new PR shortly

Reply:
`add_executor_job` expects a synchronous function so it never gets awaited. So back to my original point, don't we *want* to block the loop so we pause execution of the rest of the routine until the packages get installed? If not, imports will fail

Reply:
Yes, we definitely want to block until it completes.  However, we can't have the blocking I/O in the async function.  So just the synchronous blocking parts should go in a function called by `add_executor_job`.  See `load_files` for how it does that - it does all the I/O in single synchronous function.  In a similar way, you could have a single function that does all the file globs and file reading, and it could return a combined str, or list of the file contents, or better yet a hash indexed by file name (useful for error message reporting).  The async function could then goes ahead and parses each file's str contents and does the necessary installs.

Reply:
I see what you mean, and this may also help me fix the test. Stay tuned.